src/index.ts

```diff
@@ -1,41 +1,31 @@
 import { readdirSync } from 'fs';
-import { basename, join } from 'path';
+import { join, parse } from 'path';
 import globals from './globals.json';
 import * as snapshotProcessor from './processors/snapshot-processor';
 
 // copied from https://github.com/babel/babel/blob/d8da63c929f2d28c401571e2a43166678c555bc4/packages/babel-helpers/src/helpers.js#L602-L606
 /* istanbul ignore next */
-function interopRequireDefault(obj) {
-  return obj && obj.__esModule ? obj : { default: obj };
-}
+const interopRequireDefault = (obj: any): { default: any } =>
+  obj && obj.__esModule ? obj : { default: obj };
 
-function importDefault(moduleName) {
-  return interopRequireDefault(require(moduleName)).default;
-}
+const importDefault = (moduleName: string) =>
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  interopRequireDefault(require(moduleName)).default;
 
-const rules = readdirSync(join(__dirname, 'rules'))
-  .filter(
-    rule =>
-      rule !== '__tests__' &&
-      rule !== 'util.js' &&
-      rule !== 'tsUtils.ts' &&
-      rule !== 'tsUtils.js',
-  )
-  .map(rule =>
-    rule.endsWith('js') ? basename(rule, '.js') : basename(rule, '.ts'),
-  )
+const rulesDir = join(__dirname, 'rules');
+
+const rules = readdirSync(rulesDir)
+  .filter(rule => rule !== '__tests__' && rule !== 'utils.ts')
+  .map(rule => parse(rule).name)
```

Comment:
?

Reply:
`basename` requires you to pass (and therefore know) the file extension; `parse` doesn't, as you can just do `.name`.